DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,178 @@
+# Developer Guide
+
+The aim of this document is to give a general description of the codebase to those who would like to contribute. It will use technical language and will go deep into the various parts of the codebase.
+
+In the most general sense this is how `svelte` works.
+
+- A component is parsed into an AST
```

Comment:
Beginners might not know what an AST is, so defining the abbreviation and providing a reference could help. Also probably helpful to specify which AST. Is there a certain Typescript AST that we follow?
```suggestion
- A component is parsed into an [abstract syntax tree (AST)](https://en.m.wikipedia.org/wiki/Abstract_syntax_tree) compatible with the [ESTree spec](https://github.com/estree/estree)
```